sp/src/game/server/hl2/npc_combine.cpp

```diff
@@ -1212,6 +1333,346 @@ bool CNPC_Combine::IsGiveableItem( CBaseEntity *pItem )
+//-----------------------------------------------------------------------------
+bool CNPC_Combine::CanHeal()
+{ 
+	if ( !IsMedic() && !IsAmmoResupplier() )
+		return false;
+
+	if ( IsInAScript() || (m_NPCState == NPC_STATE_SCRIPT) )
+		return false;
+
+	return true;
+}
+
+//-----------------------------------------------------------------------------
+//-----------------------------------------------------------------------------
+bool CNPC_Combine::ShouldHealTarget( CBaseEntity *pTarget, bool bActiveUse )
+{
+	Disposition_t disposition;
+	
+	if ( pTarget && ( ( disposition = IRelationType( pTarget ) ) != D_LI && disposition != D_NU ) )
+		return false;
+
+	// Don't heal if I'm in the middle of talking
+	if ( IsSpeaking() )
+		return false;
+
+	bool bTargetIsPlayer = pTarget->IsPlayer();
+
+	// Don't heal or give ammo to targets in vehicles
+	CBaseCombatCharacter *pCCTarget = pTarget->MyCombatCharacterPointer();
+	if ( pCCTarget != NULL && pCCTarget->IsInAVehicle() )
+		return false;
+
+	if ( IsMedic() )
+	{
+		Vector toPlayer = ( pTarget->GetAbsOrigin() - GetAbsOrigin() );
+	 	if (( bActiveUse /*|| !HaveCommandGoal()*/ || toPlayer.Length() < HEAL_TARGET_RANGE) 
+			&& fabs(toPlayer.z) < HEAL_TARGET_RANGE_Z
+			)
+	 	{
+			if ( pTarget->m_iHealth > 0 )
+			{
+	 			if ( bActiveUse )
+				{
+					// Ignore heal requests if we're going to heal a tiny amount
+					float timeFullHeal = m_flPlayerHealTime;
+					float timeRecharge = sk_combine_heal_player_delay.GetFloat();
+					float maximumHealAmount = sk_combine_heal_player.GetFloat();
+					float healAmt = ( maximumHealAmount * ( 1.0 - ( timeFullHeal - gpGlobals->curtime ) / timeRecharge ) );
+					if ( healAmt > pTarget->m_iMaxHealth - pTarget->m_iHealth )
+						healAmt = pTarget->m_iMaxHealth - pTarget->m_iHealth;
+					if ( healAmt < sk_combine_heal_player_min_forced.GetFloat() )
+						return false;
+
+	 				return ( pTarget->m_iMaxHealth > pTarget->m_iHealth );
+				}
+	 				
+				// Are we ready to heal again?
+				bool bReadyToHeal = ( ( bTargetIsPlayer && m_flPlayerHealTime <= gpGlobals->curtime ) || 
+									  ( !bTargetIsPlayer && m_flAllyHealTime <= gpGlobals->curtime ) );
+
+				// Only heal if we're ready
+				if ( bReadyToHeal )
+				{
+					int requiredHealth;
+
+					if ( bTargetIsPlayer )
+						requiredHealth = pTarget->GetMaxHealth() - sk_combine_heal_player.GetFloat();
+					else
+						requiredHealth = pTarget->GetMaxHealth() * sk_combine_heal_player_min_pct.GetFloat();
+
+					if ( ( pTarget->m_iHealth <= requiredHealth ) && IRelationType( pTarget ) == D_LI )
+						return true;
+				}
+			}
+		}
+	}
+
+	// Only players need ammo
+	if ( IsAmmoResupplier() && bTargetIsPlayer )
+	{
+		if ( m_flPlayerGiveAmmoTime <= gpGlobals->curtime )
+		{
+			int iAmmoType = GetAmmoDef()->Index( STRING(m_iszAmmoSupply) );
+			if ( iAmmoType == -1 )
+			{
+				DevMsg("ERROR: Citizen attempting to give unknown ammo type (%s)\n", STRING(m_iszAmmoSupply) );
```

Comment:
This should probably say "Combine", "Combine soldier", or "soldier". Maybe even insert the classname